internal/impl/redpanda/processor_data_transforms.go

```diff
@@ -0,0 +1,414 @@
+)
+
+func dataTransformProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("Redpanda").
+		Summary("Executes a Redpanda Data Transform as a processor").
+		Description(`
+This processor executes a Redpanda Data Transform WebAssembly module, calling OnRecordWritten for each message being processed.
+`).
+		Field(service.NewStringField("module_path").
+			Description("The path of the target WASM module to execute.")).
+		Field(service.NewInterpolatedStringField("input_key").
+			Description("An optional key to populate for each message.").Optional()).
+		Field(service.NewStringField("output_key").
+			Description("An optional name of metadata for an output message key.").Optional()).
+		Field(service.NewMetadataFilterField("input_headers").
+			Description("Determine which (if any) metadata values should be added to messages as headers.").
+			Optional()).
+		Field(service.NewMetadataFilterField("output_metadata").
+			Description("Determine which (if any) message headers should be added to the output as metadata.").
+			Optional()).
+		Field(service.NewInterpolatedStringField("timestamp").
+			Description("An optional timestamp to set for each message. When left empty, the current timestamp is used.").
+			Example(`${! timestamp_unix() }`).
+			Example(`${! metadata("kafka_timestamp_unix") }`).
+			Optional().
+			Advanced()).
+		Field(service.NewDurationField("timeout").
+			Description("The maximum period of time for a message to be processed").
+			Default("10s").
+			Advanced()).
+		Field(service.NewIntField("max_memory_pages").
+			Description("The maximum amount of wasm memory pages (64KiB) that an individual wasm module instance can use").
+			Default("1600"). // 100MiB
+			Advanced()).
+		Version("0.1.0")
+}
+
+func init() {
+	err := service.RegisterBatchProcessor(
+		"redpanda_data_transform", dataTransformProcessorConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchProcessor, error) {
+			return newDataTransformProcessorFromConfig(conf, mgr)
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+//------------------------------------------------------------------------------
+
+type dataTransformConfig struct {
+	inputKey       *service.InterpolatedString
+	outputKeyField *string
+	timestamp      *service.InterpolatedString
+	inputMetadata  *service.MetadataFilter
+	outputMetadata *service.MetadataFilter
+
+	timeout        time.Duration
+	maxMemoryPages int
+}
+
+//------------------------------------------------------------------------------
+
+type dataTransformEnginePool struct {
+	log           *service.Logger
+	wasmBinary    wazero.CompiledModule
+	runtimeConfig wazero.RuntimeConfig
+	modulePool    sync.Pool
+	cfg           dataTransformConfig
+}
+
+func newDataTransformProcessorFromConfig(conf *service.ParsedConfig, mgr *service.Resources) (*dataTransformEnginePool, error) {
+	pathStr, err := conf.FieldString("module")
```

Comment:
Small nit: Not all components have been cleaned up yet, but the new pattern is to have all these field names defined as consts similar to here: https://github.com/redpanda-data/connect/blob/968786d30ca4859bda3f3c8099096d9e86feb439/internal/impl/redis/input_streams.go#L31-L41
In this case, I guess the prefix should be `dtp` for "data transform processor".

Reply:
Ack

Reply:
Done